### ISSUE TITLE
Config directory and picture directory are hardcoded

### ISSUE THREAD

**Issue by Thaodan** [anufrievroman/waypaper#15]
Waypaper hardcodes `~/.config` as directory for configuration instead of first looking into `${XDG_CONFIG_HOME}` and then falling back to `~/.config`.

The situation is similar for the pictures directory which
would require to parse `${XDG_CONFIG_HOME}/user-dirs.dirs`
to get `XDG_PICTURES_DIR`.

The platformdirs [module](https://github.com/platformdirs/platformdirs) should be useful for both cases.

You can append a pathlib object with a string by doing:
`path / string` e.g. `path / foo.exist()`.

**Comment by anufrievroman** [comment]
Good idea 👍 Do you think some Linux systems have it not in `~/.config`? Currently, I am a bit occupied but I'll try to look into it.

**Comment by Thaodan** [comment]
> Good idea 👍 Do you think some Linux systems have it not in `~/.config`? 

On my system I did set it to `~/.local/etc` to have everything in `~/.local ` to only have 
dot-folder, that's how I noticed it. But usually that's not the case, during testing it can be handy thou to set these directories to something else.

> Currently, I am a bit occupied but I'll try to look into it.

I worked on it last night, including fixing some other bugs such as that waypaper doesn't work if GdxPixbuf can't open an image. Only issue is that the index still counts the files that shouldn't be decoded, either not show them at all or use a placeholder.

The branch can be found here but it's not finished:
https://github.com/thaodan/waypaper/tree/xdg_dirs
